src/extent.c

```diff
@@ -908,6 +908,9 @@ extent_try_coalesce_impl(tsdn_t *tsdn, pac_t *pac, ehooks_t *ehooks,
 		edata_t *next = emap_try_acquire_edata_neighbor(tsdn, pac->emap,
 		    edata, EXTENT_PAI_PAC, ecache->state, /* forward */ true);
 		if (next != NULL) {
+			if ((max_size != 0) && ((edata_size_get(edata) + edata_size_get(next)) > max_size)) {
```

Comment:
Need to consider overflow here. Maybe:
```
size_t max_next_neighbor = max_size > edata_size_get(edata) ?  max_size - edata_size_get(edata) : 0;
if (next != NULL && edata_size_get(next) <= max_next_neighbor) {
...
...
size_t max_prev_neighbor = max_size > edata_size_get(edata) ?  max_size - edata_size_get(edata) : 0;
if (prev != NULL && edata_size_get(prev) <= max_prev_neighbor) {
...
```

(also see the next comment about not using 0 to encode size_max)